### PR TITLE
Add OpenCV 3.3.1

### PR DESCRIPTION
**Pull request by xsacha**

### CODE REVIEW

**Comment by ruslo** [comment]
Test OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2731
* https://travis-ci.org/ingenue/hunter/builds/300069380?utm_source=github_status&utm_medium=notification

Test OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2732
* https://travis-ci.org/ingenue/hunter/builds/300069515?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> Test OpenCV

Ninja and NMake toolchains stop working, please exclude.

> Test OpenCV-extra

All tests failed. It seems that old version is not compatible with 3.3.1. Can you please update OpenCV-extra package too.

**Comment by xsacha** [comment]
Ninja and NMake fail to build the new DNN module (new in 3.3.0). Not sure whether this module should be disabled or to remove testing for Ninja and NMake.

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -93,7 +93,7 @@ hunter_config(OpenAL VERSION 1.18.2)
 hunter_config(OpenBLAS VERSION 0.2.20-p0)
 hunter_config(OpenCL VERSION 2.1-p3)
 hunter_config(OpenCL-cpp VERSION 2.0.10-p0)
-hunter_config(OpenCV VERSION 3.3.0-p2)
+hunter_config(OpenCV VERSION 3.3.1-p0)
 hunter_config(OpenCV-Extra VERSION 3.3.0)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Rebase change and update version of `OpenCV-Extra` too.

**Comment by ruslo** [comment]
Test OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2768
* https://travis-ci.org/ingenue/hunter/builds/300564322?utm_source=github_status&utm_medium=notification

Test OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2769
* https://travis-ci.org/ingenue/hunter/builds/300564497?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Some iOS toolchains need to be checked locally. Will do tests and report back when done.

**Comment by ruslo** [comment]
> Some iOS toolchains need to be checked locally. Will do tests and report back when done.

Works fine.

**Comment by ruslo** [comment]
OpenCV upload:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2779
* https://travis-ci.org/ingenue/hunter/builds/300852495?utm_source=github_status&utm_medium=notification

OpenCV-extra upload:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2780
* https://travis-ci.org/ingenue/hunter/builds/300852548?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.151

**Pull request merged**